src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -131,7 +137,10 @@ public bool IsEmpty(DocumentId documentId)
 
                 if (!await TryDeserializeDocumentAsync(serializer, document, builder, cancellationToken).ConfigureAwait(false))
                 {
-                    Contract.Requires(false, "How this can happen?");
+                    Contract.Requires(lastResult.Version == VersionStamp.Default);
```

Comment:
I haven't merged the test yet. I can wait for your PR to get merged, and then checkin the test.